File: 1817-calculate-money-in-leetcode-bank/calculate-money-in-leetcode-bank.cpp
class Solution {
public:
    int totalMoney(int n) {
        int x =1 ;
        int ans = 0 ;
        int days = 1 ;
        int week = 1 ;
        while(n--){
            ans += x ;
            x++ ;
            
            if(days%7 == 0){
                week+=1 ;
                x = week ;
            }
            days++ ;
        }
        return ans ;
    }
};

File: 1817-calculate-money-in-leetcode-bank/calculate-money-in-leetcode-bank.py
class Solution:
    def totalMoney(self, n: int) -> int:
        ans = 1
        days = 1
        amt = 1
        week = 1

        while(n):
            ans+= amt 
            amt += 1

            if(days %7 == 0):
                week += 1 
                amt = week
            
            days +=1 
            n -= 1
        return ans-1 
